libtenzir/builtins/aggregation-functions/mean.cpp

```diff
@@ -68,7 +70,83 @@ class mean_function final : public aggregation_function {
   size_t count_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class mean_instance final : public aggregation_instance {
+public:
+  explicit mean_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    // TODO: Collapse them into a generic, or think about compute api
+    // TODO: Actually calculate the mean
+    auto f = detail::overload{
+      [](const arrow::NullArray&) {},
+      [&](const arrow::Int64Array& array) {
+        for (auto i = int64_t{}; i < array.length(); ++i) {
+          if (array.IsValid(i)) {
+            const auto val = array.Value(i);
+            if (not mean_ || val < std::get<int64_t>(mean_.value())) {
+              mean_ = val;
+            }
+          }
+        }
+      },
```

Comment:
This does not look like a `mean` function to me.

Reply:
Yep, there was a TODO at top. Added.